bot/tests/test_index.py

```diff
@@ -0,0 +1,252 @@
+    assert args["data"]["test"] == "dummy"
+    assert args["data"]["id"] == "1234"
+    assert args["data"]["source"] == "try"
+    assert args["data"]["try_task_id"] == "remoteTryTask"
+    assert args["data"]["try_group_id"] == "remoteTryGroup"
+    assert args["data"]["repository"] == "test-repo"
+    assert args["data"]["someData"] == "mock"
+    assert "indexed" in args["data"]
+
+    # Second call with sub namespace
+    namespace, args = calls[1][0]
+    assert namespace == "project.relman.test.code-review.mock.1234.12345deadbeef"
+    assert args["taskId"] == "12345deadbeef"
+    assert args["data"]["test"] == "dummy"
+    assert args["data"]["id"] == "1234"
+    assert args["data"]["source"] == "try"
+    assert args["data"]["try_task_id"] == "remoteTryTask"
+    assert args["data"]["try_group_id"] == "remoteTryGroup"
+    assert args["data"]["repository"] == "test-repo"
+    assert args["data"]["someData"] == "mock"
+    assert "indexed" in args["data"]
+
+
+def test_index_autoland(
+    mock_autoland_task,
+    mock_phabricator,
+    mock_hgmo,
+    mock_config,
+    mock_workflow,
+    mock_taskcluster_date,
+):
+    """
+    Check the Taskcluster index generated for an autoland task
+    """
+
+    with mock_phabricator as api:
+        revision = Revision.from_decision_task(mock_autoland_task, api)
+
+    mock_workflow.index_service = mock.Mock()
+    mock_config.taskcluster = TaskCluster("/tmp/dummy", "12345deadbeef", 0, False)
+
+    mock_workflow.index(revision, state="unit-test")
+
+    assert mock_workflow.index_service.insertTask.call_count == 2
+    calls = mock_workflow.index_service.insertTask.call_args_list
+
+    assert [c[0][0] for c in calls] == [
+        "project.relman.test.code-review.head_repo.integration-autoland.deadbeef123",
+        "project.relman.test.code-review.head_repo.integration-autoland.deadbeef123.12345deadbeef",
+    ]
+
+    # Check all calls have the same shared payload
+    payload = {
+        "base_changeset": "123deadbeef",
+        "base_repository": "https://hg.mozilla.org/mozilla-unified",
+        "bugzilla_id": None,
+        "diff_id": None,
+        "diff_phid": None,
+        "has_clang_files": False,
+        "head_changeset": "deadbeef123",
+        "head_repository": "https://hg.mozilla.org/integration/autoland",
+        "id": None,
+        "indexed": "2025-10-30T00:00:00.00Z",
+        "mercurial_revision": "deadbeef123",
+        "monitoring_restart": False,
+        "phid": None,
+        "repository": "https://hg.mozilla.org/mozilla-unified",
+        "source": "try",
+        "state": "unit-test",
+        "target_repository": "https://hg.mozilla.org/mozilla-unified",
+        "title": "Changeset deadbeef123 (https://hg.mozilla.org/integration/autoland)",
+        "try_group_id": "remoteTryGroup",
+        "try_task_id": "remoteTryTask",
+        "url": None,
+    }
+    assert calls[0][0][1]["data"] == payload
+    assert all([c[0][1]["data"] == payload for c in calls])
```

Comment:
Based on line 97, calls is not empty. Should line 127 stand alone and line 126 be dropped because 127 covers it? Similar below.